src/MSBuild.Abstractions/LooseVersionAssemblyLoader.cs

```diff
@@ -76,7 +76,7 @@ private static Assembly TryResolveAssemblyFromPaths(AssemblyLoadContext context,
         private static Assembly LoadAndCache(AssemblyLoadContext context, string fullPath)
         {
             var assembly = context.LoadFromAssemblyPath(fullPath);
-            var name = assembly.FullName;
+            var name = AssemblyLoadContext.GetAssemblyName(fullPath).FullName;
```

Comment:
why this change?

Reply:
My logic was:
`context.LoadFromAssemblyPath(fullPath);` returns `Assembly` instance in which `string? FullName` is nullable
whereas `AssemblyLoadContext.GetAssemblyName(fullPath)` returns `AssemblyName` instance where `string FullName` is marked as non-nullable.

Shall I revert the change?

Reply:
I think its ok to keep it. I'll ask the CoreFx folks what horrible back-compat story they must have for why `Assembly.FullName` can return null but `AssemblyLoadContext.GetAssemblyName("").FullName` cannot. I assume that `AssemblyLoadContext.GetAssemblyName` throws if it cannot file the assembly name where as `Assembly.FullName` returns null and they need to keep this around for backcompat.